snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -357,27 +359,29 @@ def _get_code_name_from_build_base(self):
             "core20": "focal",
         }[build_base]
 
-    def _get_primary_mirror(self) -> str:
+    def _get_environment_arch(self) -> str:
+        arch = self._run(["arch"], hide_output=True)
```

Comment:
won't this print the machine hardware? what we want is the userspace architecture though, right?

Reply:
indeed. will revisit.

since I can't rely on `dpkg-architecture` do you have a better suggestion?

Reply:
Though in this case, it's really just used to check for x86 or not, so should be safe.  At least as safe as platform.machine().

Reply:
we have something for user space architecture checking in project_options. I would add a code comment to get the intent as _get_environment_arch is not returning an accurate value if this caveat is taken into account